### PR TITLE
perf(core): allocation-free is_instruction_file and reduced PathBuf clones

### LINKED ISSUE #470: Performance: reduce PathBuf clones in parallel file walk
## Audit Finding (Performance, Medium)

**Location**: `crates/agnix-core/src/pipeline.rs:799-806`

In the parallel validation fold, `file_path` is cloned for AGENTS.md and instruction file checks on every file, even when the conditions are false. For 10,000 files this wastes ~1MB of allocations.

**Fix**: Only clone when the condition is true (the clones are already conditional, but the clone happens before the push). Verify the current code is actually cloning unconditionally or if the compiler optimizes this away.

Found by `/audit-project`

### PR DESCRIPTION
**Pull request by avifenesh**
## Summary
- Rewrote `is_instruction_file()` to use `eq_ignore_ascii_case` and `Path::components()` instead of allocating `to_string_lossy().to_lowercase()` strings - eliminates 2 heap allocations per file during project validation walks
- Eliminated PathBuf clone on error path in the parallel validation fold by moving the owned value into the diagnostic
- Fixes a latent false-positive bug where `path_str.contains(".cursor")` would match filenames like `my.cursor-notes.mdc`

## Test plan
- [x] All 2891 existing tests pass (`cargo test --package agnix-core`)
- [x] 7 new edge case tests added for `is_instruction_file()` covering case variations, false positives, deep nesting, empty/absolute/relative paths
- [x] New criterion benchmark for `is_instruction_file()` with 20 representative paths
- [x] Clippy clean (`cargo clippy --package agnix-core`)
- [x] Full workspace builds

Closes #470

### CODE REVIEW

File: crates/agnix-core/src/schemas/cross_platform.rs [hunk 1]
```diff
@@ -945,34 +945,104 @@ pub fn detect_precedence_issues(layers: &[InstructionLayer]) -> Option<LayerPrec
...
+            _ => continue,
+        };
+
+        if s.eq_ignore_ascii_case(".cursor") {
+            found_cursor = true;
+        } else if s.eq_ignore_ascii_case("rules") {
+            found_rules = true;
+        } else if s.eq_ignore_ascii_case(".github") {
+            found_github = true;
+        } else if found_github && ascii_contains_ignore_case(s, "copilot") {
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply crates/agnix-core/src/schemas/cross_platform.rs:1005]: This flag is scoped to a single is_instruction_file() call - the loop iterates components of one path, not multiple paths. Each call starts with all flags at false, so there is no cross-path contamination. The flag correctly captures whether a copilot-related component appears after .github in this path.

File: crates/agnix-core/src/schemas/cross_platform.rs [hunk 2]
```diff
@@ -945,34 +945,104 @@ pub fn detect_precedence_issues(layers: &[InstructionLayer]) -> Option<LayerPrec
...
+    if found_opencode {
+        return true;
+    }
+
+    false
+}
+
+/// Case-insensitive ASCII substring search without allocating.
+fn ascii_contains_ignore_case(haystack: &str, needle: &str) -> bool {
+    if needle.is_empty() || needle.len() > haystack.len() {
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply crates/agnix-core/src/schemas/cross_platform.rs]: Fixed - changed to return `true` for empty needle to match stdlib `str::contains` semantics, and split into separate guard clauses for clarity.

File: crates/agnix-core/src/schemas/cross_platform.rs [hunk 3]
```diff
@@ -945,34 +945,104 @@ pub fn detect_precedence_issues(layers: &[InstructionLayer]) -> Option<LayerPrec
...
+        } else if s.eq_ignore_ascii_case(".opencode") {
+            found_opencode = true;
+        }
+    }
+
+    // .cursor directory: filename ends with .mdc OR any component is "rules"
+    if found_cursor {
+        let has_mdc_ext = file_name
+            .rsplit('.')
+            .next()
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply crates/agnix-core/src/schemas/cross_platform.rs]: Fixed - switched to `Path::new(file_name).extension()` which correctly returns `None` for extensionless filenames like `mdc`, preventing the false positive.

File: crates/agnix-core/src/schemas/cross_platform.rs [hunk 4]
```diff
@@ -945,34 +945,104 @@ pub fn detect_precedence_issues(layers: &[InstructionLayer]) -> Option<LayerPrec
...
+    }
+
+    false
+}
+
+/// Case-insensitive ASCII substring search without allocating.
+fn ascii_contains_ignore_case(haystack: &str, needle: &str) -> bool {
+    if needle.is_empty() || needle.len() > haystack.len() {
+        return false;
+    }
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply crates/agnix-core/src/schemas/cross_platform.rs]: Already fixed in commit 833dc97 - the empty needle now returns true to match stdlib semantics.
